src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const CloneObject = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(CloneObject, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete CloneObject[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in CloneObject) {
+          delete CloneObject[key];
+        }
+        break;
+    }
```

Comment:
Consider using a 'default' case in your switch statement to handle any unexpected 'action.type' values. This can help with error handling and ensure that the function behaves correctly even if it receives an unknown action.